src/mgmt/srv_drpc.c

```diff
@@ -747,6 +747,127 @@ ds_mgmt_drpc_pool_get_acl(Drpc__Call *drpc_req, Drpc__Response *drpc_resp)
 	free_ace_list(ace_list, ace_nr);
 }
 
+void
+ds_mgmt_drpc_list_pools(Drpc__Call *drpc_req, Drpc__Response *drpc_resp)
+{
+	Mgmt__PoolListReq		*req = NULL;
+	Mgmt__PoolListResp		*resp = NULL;
```

Comment:
You can actually allocate this one on the stack - eliminates some memory management drudgery.
Something like:
`Mgmt__PoolListResp resp = MGMT__POOL_LIST_RESP__INIT;`

You won't need to free the resp itself at that point, just the contents of the pool array.